Monofoxe/Monofoxe.Engine/Utils/GameMath.cs

```diff
@@ -208,24 +208,124 @@ public static float GetArea(List<Vector2> vertices)
 		}
 
 
+
+		#region Bezier curve stuff.
+		// A look up table for factorials. Capped to 16.
+		private static float[] Factorial = new float[]
+		{
+			1.0f,
+			1.0f,
+			2.0f,
+			6.0f,
+			24.0f,
+			120.0f,
+			720.0f,
+			5040.0f,
+			40320.0f,
+			362880.0f,
+			3628800.0f,
+			39916800.0f,
+			479001600.0f,
+			6227020800.0f,
+			87178291200.0f,
+			1307674368000.0f,
+			20922789888000.0f,
+		};
+
+		private static float Binomial(int n, int i)
+		{
+			float ni;
+			float a1 = Factorial[n];
+			float a2 = Factorial[i];
+			float a3 = Factorial[n - i];
+
+			ni = a1 / (a2 * a3);
+
+			return ni;
+		}
+
+		private static float Bernstein(int n, int i, float value)
+		{
+			float value_i = MathF.Pow(value, i);
+			float value_n_minus_i = MathF.Pow((1 - value), (n - i));
+
+			float basis = Binomial(n, i) * value_i * value_n_minus_i;
+
+			return basis;
+		}
+		#endregion
+
+
 		/// <summary>
-		/// Calculates three-point bezier curve.
+		/// Calculates bezier curve.
 		/// </summary>
 		/// <param name="value">Should be in 0..1 range.</param>
-		public static Vector2 BezierCurve(Vector2 startPoint, Vector2 controlPoint, Vector2 endPoint, float value)
+		public static Vector2 BezierCurve(Vector2[] controlPoints, float value)

```

Comment:
Return three-point BezierCurve, it can exist alongside the n-point bezier. 
```
		public static Vector2 BezierCurve(float value, params Vector2[] points)
		{ 
// ...
		}

		public static Vector2 BezierCurve(float value, Vector2 startPoint, Vector2 controlPoint, Vector2 endPoint)
		{
// ...
		}
```